### PR TITLE
Remove STATUS file from release

### PR DESCRIPTION
**Pull request by daviditen**
Stop building STATUS in gen_release and stop listing it as a file to include.

Remove STATUS from README.devel, README.files, QUICKSTART.rst, chpl man page,
and chpldoc man page.

Point some compiler warning messages at a GitHub issue instead of the
STATUS file.  Update a few .good files to match the updated warnings.

Change bugs.rst to point at STATUS.devel on GitHub instead of STATUS for
issues pre-dating the move to GitHub issues.

Point to the GitHub Issues list instead of STATUS in GOALS.md

### CODE REVIEW

File: man/chpl.rst [hunk 1]
```diff
@@ -741,7 +741,6 @@ effect as passing that option once.
 BUGS
 ----
 
-See $CHPL\_HOME/STATUS for a list of known bugs and
 $CHPL\_HOME/doc/bugs.rst for instructions on reporting bugs.
```
Comment (awallace-cray) [inline man/chpl.rst]: "See" $CHPL\_HOME/doc/bugs.rst for instructions on reporting bugs.
?

File: man/chpldoc.rst [hunk 2]
```diff
@@ -126,7 +126,6 @@ effect as passing that option once.
 BUGS
 ----
 
-See $CHPL\_HOME/STATUS for a list of known bugs and
 $CHPL\_HOME/doc/bugs.rst for instructions on reporting bugs.
```
Comment (awallace-cray) [inline man/chpldoc.rst]: "See" $CHPL_HOME/doc/bugs.rst for instructions on reporting bugs.
?
(same comment as chpl.rst file)

File: GOALS.md [hunk 3]
```diff
@@ -25,4 +25,4 @@ The goals of this release are as follows:
        domains and arrays using some standard distributions.
 
 For more detailed notes on what is or is not supported in this
-release, refer to the STATUS file in this directory.
+release, refer to the Issues list on GitHub.
```
Comment (awallace-cray) [inline GOALS.md]: Tell them where the Issues list on GitHub is? (https://github.com/chapel-lang/chapel/issues)

**Comment by awallace-cray** [comment]
Above comments are wordsmith details only, please consider them as suggestions. Everything else looks good to me.  J'approve.

**Comment by daviditen** [comment]
Thanks @awallace-cray !  I made the edits you suggested.

**Comment by bradcray** [comment]
Looks like you may have missed:
* the .gitignore file
* Makefile.devel

**Comment by daviditen** [comment]
I removed STATUS from the release, but didn't completely remove our ability to create it - in fact it still gets created when you do a top-level "make".  Do you prefer to see it removed completely?

**Comment by bradcray** [comment]
I see...  I was thinking that there probably wasn't any reason to preserve the ability to build it since the STATUS.devel is more useful for developers (points to the tests in question).

I also hadn't noticed that this had been merged before I commented on it...

**Comment by daviditen** [comment]
I opened a new PR #5772 to remove STATUS from Makefile.devel and from .gitignore. @bradcray , would you like to look at that one?

**Pull request merged**